packages/runtime/datastore-definitions/src/dataStoreRuntime.ts

```diff
@@ -76,6 +76,8 @@ export interface IFluidDataStoreRuntime
 	 */
 	readonly attachState: AttachState;
 
+	get inStagingMode(): boolean;
```

Comment:
I don't like this addition - fine for prototyping purposes but should make a plan to keep the source of truth in one place for productizing.

Example might be to have the container runtime factory inject the superpower into the StressDataObject factory as it places it in the registry, so StressDataObjects can be given the power as they are created.

Observable, changeable state probably also wants an eventing story long-term.

Reply:
we'll see. we follow a similar pattern with a bunch or our states, attach, connected, readonly, etc where we project them down. i have a follow up change that need this as well, and i could see data store authors needing it to.

the shape here isn't final in the least, and agreed we may need some kind of eventing story

Reply:
I think it will be important for datastores to know if they are operating in staging mode, as they may do things like visualize or track data differently in staging mode. i agree we should limit the ability to enter or exit the mode.

Reply:
For now probably good to explore all paths.  However, I think as a platform we need to clarify the mental model of staging mode, in particular whether it's a "container state" or a "datastore state".  I worry that if the API is across both just for convenience purposes it makes the mental model murkier.

And as mentioned it's easy for customers to inject any container-level stuff down to datastores at their discretion, putting the control in their hands.